cli/cmd/sync_v3.go

```diff
@@ -50,10 +50,19 @@ func syncConnectionV3(ctx context.Context, sourceClient *managedplugin.Client, d
 		}
 		if destinationSpecs[i].WriteMode == specs.WriteModeAppend {
 			opts = append(opts, transformer.WithRemovePKs())
+		} else if destinationSpecs[i].PKMode == specs.PKModeCQID {
```

Comment:
```suggestion
		}
		if destinationSpecs[i].PKMode == specs.PKModeCQID {
```

Reply:
Hmm, I don't think this is equivalent. If write mode is append, we should remove all PKs and not add any back (not even `_cq_id`). If write mode is not append, we should check the pk mode and replace all PKs with CQID only if pk_mode == cq_id. This seems right to me, and it's what the logic was doing before

Reply:
OK makes sense now that you've explained it in detail :)